builder/dockerfile/dispatchers.go

```diff
@@ -293,7 +292,7 @@ func workdir(b *Builder, args []string, attributes map[string]bool, original str
 //
 func run(b *Builder, args []string, attributes map[string]bool, original string) error {
 	if b.image == "" && !b.noBaseImage {
-		return derr.ErrorCodeMissingFrom
+		return fmt.Errorf("Please provide a source image with `from` prior to run")
```

Comment:
@WeiZhang555 @mlaventure I agree that we should find a compromise, but I rather not make more changes than necessary in this PR. It's already big enough as it is.